### PR TITLE
3272-Slot-Demo-HistorySlot

### LINKED ISSUE #3272: Slot Demo: HistorySlot
add a demo: History slot.

It saves 5 old values.

### PR DESCRIPTION
**Pull request by MarcusDenker**
fixes #3272

Implementation of HistorySlot and tests

### CODE REVIEW

**Comment by MarcusDenker** [comment]
failing tests not related

File: src/Slot-Tests/HistorySlotTest.class.st [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+	slot := #slot1 => HistorySlot.
+	aClass := self make: [ :builder | builder slots: {slot} ].
+
+	self compileAccessorsFor: slot.
+
+	instance := aClass new.
+	instance slot1: 25.
+	instance slot1: 35.
+	self assert: instance slot1 equals: 35.
+	self assert: (slot readHistory: instance) size equals: 3
```
Comment (bencoman) [inline src/Slot-Tests/HistorySlotTest.class.st]: #size seems a softer condition than an array of the historic values, and also demonstrate how it works.
Is it #(25 35 35) or #(35 35 25) ?

Comment (MarcusDenker) [reply src/Slot-Tests/HistorySlotTest.class.st]: yes, I made it better... especially as the first value is nil, which is not at all that obvious

**Comment by bencoman** [comment]
nice. thx.

**Pull request merged**